watertap/unit_models/leaky_pressure_exchanger.py

```diff
@@ -0,0 +1,491 @@
+        domain=In(MomentumBalanceType),
+        description="Momentum balance construction flag",
+        doc="""Indicates what type of momentum balance should be constructed,
+    **default** - MomentumBalanceType.pressureTotal.
+    **Valid values:** {
+    **MomentumBalanceType.none** - exclude momentum balances,
+    **MomentumBalanceType.pressureTotal** - single pressure balance for material,
+    **MomentumBalanceType.pressurePhase** - pressure balances for each phase,
+    **MomentumBalanceType.momentumTotal** - single momentum balance for material,
+    **MomentumBalanceType.momentumPhase** - momentum balances for each phase.}"""))
+    CONFIG.declare("property_package", ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description="Property package to use for control volume",
+        doc="""Property parameter object used to define property calculations,
+        **default** - useDefault.
+        **Valid values:** {
+        **useDefault** - use default package from parent model or flowsheet,
+        **PhysicalParameterObject** - a PhysicalParameterBlock object.}"""))
+    CONFIG.declare("property_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing property packages",
+        doc="""A ConfigBlock with arguments to be passed to a property block(s)
+        and used when constructing these,
+        **default** - None.
+        **Valid values:** {
+        see property package for documentation.}"""))
+
+    def build(self):
+        super().build()
+
+        # Pressure exchanger supports only liquid phase
+        if self.config.property_package.phase_list != ['Liq']:
+            raise ConfigurationError(
+                "Pressure exchanger model only supports one liquid phase ['Liq'],"
+                "the property package has specified the following phases {}"
+                    .format(self.config.property_package.phase_list))
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        self.efficiency_pressure_exchanger = Var(
+            self.flowsheet().config.time,
+            initialize=0.95,
+            bounds=(1e-6, 1),
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            doc='Pressure exchanger efficiency')
+
+        self.solute_leakage_fraction = Var(
+            self.flowsheet().config.time,
+            initialize=0.05,
+            bounds=(1e-6, 1),
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            doc='Pressure exchanger solute leakage')
+
+        self.solvent_leakage_fraction = Var(
+            self.flowsheet().config.time,
+            initialize=0.05,
+            bounds=(1e-6, 1),
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            doc='Pressure exchanger solvent leakage')
```

Comment:
I would of thought that there should just be one leakage fraction because isn't it the whole solution that leaks? Or is there some preferential mechanism that has more solvent or solute leak like diffusion?

Reply:
This depends on design of pressure exchanger. In some of them you might have simply diffusion or mixing happing in the exchange volume so you only have transfer of solute, but now liquid mass transfer to account for (or it might be very minimal). 
Check out this design (https://www.youtube.com/watch?v=mJe1nynkBGk) , the low pressure and high pressure fluid come in direct contact and there is no "piston" to separate them, the two fluids mix, and can exit evenly such that you have only solute transport but minimal or on water transport.

Reply:
Explained for cases where there convective/diffusion mixing of high and low pressure fluid in pressure exchange model.

Reply:
@avdudchenko you could also just have `self.leakage_fraction_comp` and index by the `component_list` which would include solvent and solute (we generally use the qualifier "comp" for anything that is indexed by component.) I didn't thoroughly review all of your code yet, but as @TimBartholomew mentioned in another comment, and assuming that these leakage fractions are the only new variables added to the pressure exchanger, you could declare a new boolean configuration option such as `has_leakage`; if `True`, create the leakage fraction vars (and later on, apply that `has_leakage` check to the associated constraints).

Reply:
Yes I think that is my plan at this point.